### PR TITLE
feat: track btc fees on success

### PR DESCRIPTION
**Pull request by kylezs**
# Pull Request

Closes: PRO-1034

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works - did test the recorded fee equals that of the local explorer, and it matches.
- [x] I have updated documentation where appropriate.

## Summary

We want to track how much fees the vault has used to pay for egress.

### CODE REVIEW

**Comment by AlastairHolmes** [comment]
This waiting for something?

**Comment by kylezs** [comment]
@martin-chainflip was going to review

**Review by martin-chainflip — APPROVED** [review]

File: engine/src/btc/rpc.rs [hunk 1]
```diff
@@ -208,37 +210,120 @@ async fn get_bitcoin_network(
...
 #[serde(rename_all = "camelCase")]
 pub struct BlockHeader {
 	pub hash: bitcoin::BlockHash,
 	pub confirmations: i32,
 	pub height: u64,
 	pub version: Version,
+	pub version_hex: Option<String>,
 	// Don't care to write custom deserializer for this
-	#[serde(skip)]
-	pub version_hex: Option<Vec<u8>>,
```
Comment (AlastairHolmes) [inline engine/src/btc/rpc.rs:220]: Why did you add this?

Comment (AlastairHolmes) [reply engine/src/btc/rpc.rs:220]: Same below

File: engine/src/btc/rpc.rs [hunk 2]
```diff
@@ -208,37 +210,120 @@ async fn get_bitcoin_network(
...
+	struct Helper {
+		hex: String,
+	}
+	let helper = Helper::deserialize(deserializer)?;
+	Ok(ScriptBuf::from(hex::decode(helper.hex).map_err(serde::de::Error::custom)?))
+}
+
+#[derive(Clone, PartialEq, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum Difficulty {
```
Comment (AlastairHolmes) [inline engine/src/btc/rpc.rs:254]: So this was broken before?

Comment (AlastairHolmes) [reply engine/src/btc/rpc.rs:254]: The decoding I mean
